### ISSUE TITLE
Push receiving after TTL time 

### ISSUE THREAD

**Issue by SandeepKhore** [AndrewBarba/apns2#51]
We have observed that in multiple occurrences we are receiving push after the TTL expiry. We have set the TTL expiry to 60 seconds but still we are receiving push after 15 minutes.

Let us know what details you required in order to fix this issue.

Please find our code snippet below for your reference.

`const client = new APNS({
  host: production ? hostProduction : hostDevelopment,
  port,
  team: apnTeamId,
  keyId: apnKeyId,
  signingKey: fs.readFileSync(apnPath),
  defaultTopic: topic,
});`

`const now = new Date();
    let futureTime = now.setSeconds(now.getSeconds() + nonVoipPushExpiration);
    const expireDate = new Date(futureTime);
    
    notification = new Notification(deviceToken.token, {
      data,
      pushType: NotificationType.PushType.voip,
      priority: 10,
      expiration: expireDate,
    });
    
    return client.send(notification);`

**Comment by SandeepKhore** [comment]
@AndrewBarba  anything on this

**Comment by AndrewBarba** [comment]
Take a look at Apples explanation here: https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/sending_notification_requests_to_apns

60s is a tight window. Instead if setting an expiration date try passing 0 for expiration. This will attempt to deliver the notification only once.

**Comment by SandeepKhore** [comment]
Yeah, did try this but in some cases we were still loosing the pushes so added some extra notification methodologies using socket and MQTT to notify device in case device is in foreground and if device is in deep sleep mode or not connected to socket/MQTT then sending a non-voip push to awake device and then sending voip push.

By this approach we are now able to receive 98% push notification.
